ml-proto/spec/check.ml

```diff
@@ -392,15 +429,11 @@ let check_memory_segment c prev_end seg =
 
 let check_global c g =
   let {gtype; value} = g.it in
-  check_const c gtype value
+  check_const c value gtype
 
 let check_start c start =
   Lib.Option.app (fun x ->
-    let start_type = func c x in
-    require (start_type.ins = []) x.at
-      "start function must be nullary";
-    require (start_type.out = None) x.at
-      "start function must not return anything";
+    require (func c x = FuncType ([], [])) x.at "start function must be nullary"
```

Comment:
This change causes the interpreter to say "start function must be nullary" when given a nullary function that has a return value, which is confusing.